spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/AbstractMqttMessageHandler.java

```diff
@@ -86,10 +90,20 @@ public abstract class AbstractMqttMessageHandler extends AbstractMessageHandler
 
 	private int clientInstance;
 
+	private final ClientManager<T> clientManager;
+
 	public AbstractMqttMessageHandler(@Nullable String url, String clientId) {
 		Assert.hasText(clientId, "'clientId' cannot be null or empty");
 		this.url = url;
 		this.clientId = clientId;
+		this.clientManager = null;
+	}
+
+	AbstractMqttMessageHandler(ClientManager<T> clientManager) {
```

Comment:
Why this one is not `public`?

Reply:
As far as I remember, it was one of Sonar default rules, not to make abstract classes constructors public.

Reply:
Sounds weird. How else I suppose to extend this class in some other place outside of this package?
Here is what we have so far in our SonarQube: https://sonar.spring.io/dashboard?id=org.springframework.integration%3Aspring-integration%3Amain

Reply:
Regarding that default Sonar rule, I guess that protected or package-private visibility was meant to be used in abstract classes rather than public. You can always use protected constructor in the impl classes.
I tried to setup my SonarLint plugin to look at your sonar cloud, but apparently I should be spring memeber 🙂 So my apologies, our rules are a bit different.
Anyway. Should I then just make it public or protected is OK to you?

Reply:
I guess `public`  to be consistent with existing ctors in this class.
I don't personal preference about this though, but better to have it as flexible as possible for end-users.
I guess our Sonar rule on the matter is disabled 😄 .